autofaiss/external/quantize.py

```diff
@@ -20,278 +21,273 @@
 from autofaiss.indices.index_utils import set_search_hyperparameters, load_index_from_hdfs, save_index_on_hdfs
 from autofaiss.utils.decorators import Timeit
 from autofaiss.utils.cast import cast_memory_to_bytes, cast_bytes_to_memory_string
-
-
-class Quantizer:
-    """class defining the quantization pipeline"""
-
-    def __init__(self):
-        """Empty constructor"""
-
-    def quantize(
-        self,
-        embeddings_path: str,
-        output_path: str,
-        index_key: Optional[str] = None,
-        index_param: Optional[str] = None,
-        max_index_query_time_ms: float = 10.0,
-        max_index_memory_usage: str = "16G",
-        current_memory_available: str = "32G",
-        use_gpu: bool = False,
-        metric_type: str = "ip",
-        nb_cores: Optional[int] = None,
-    ) -> str:
-        """
-        Reads embeddings and creates a quantized index from them.
-        The index is stored on the current machine at the given ouput path.
-
-        Parameters
-        ----------
-        embeddings_path : str
-            Local path containing all preprocessed vectors and cached files.
-            Files will be added if empty.
-        output_path: str
-            Destination path of the quantized model on local machine.
-        index_key: Optinal(str)
-            Optional string to give to the index factory in order to create the index.
-            If None, an index is chosen based on an heuristic.
-        index_param: Optional(str)
-            Optional string with hyperparameters to set to the index.
-            If None, the hyper-parameters are chosen based on an heuristic.
-        max_index_query_time_ms: float
-            Bound on the query time for KNN search, this bound is approximative
-        max_index_memory_usage: str
-            Maximum size allowed for the index, this bound is strict
-        current_memory_available: str
-            Memory available on the machine creating the index, having more memory is a boost
-            because it reduces the swipe between RAM and disk.
-        use_gpu: bool
-            Experimental, gpu training is faster, not tested so far
-        metric_type: str
-            Similarity function used for query:
-                - "ip" for inner product
-                - "l2" for euclidian distance
-        nb_cores: Optional[int]
-            Number of cores to use. Will try to guess the right number if not provided
-        """
-
-        current_bytes = cast_memory_to_bytes(current_memory_available)
-        max_index_bytes = cast_memory_to_bytes(max_index_memory_usage)
-        memory_left = current_bytes - max_index_bytes
-
-        if memory_left < current_bytes * 0.1:
+import numpy as np
+
+
+def quantize(
+    embeddings_path: str,
```

Comment:
Can you change name to embeddings_or_path and type to Union[str, np.ndarray] ?

Reply:
yes done

Reply:
I changed the type but actually I think it's better if we don't change the argument name, it would be an annoying breaking change with no obvious benefit